### DISCUSSION TITLE
UE4 4.26 engine for Carla 0.9.12 building error for Windows build in Visual Studios 19

### DISCUSSION THREAD

**Discussion by lallard1** [carla-simulator/carla#5496] [Q&A]
I'm attempting step 3 of the Unreal Engine portion of the Windows build (https://carla.readthedocs.io/en/latest/build_windows/). 
The build gets to the final step but fails with the following error message (this was from a rebuild attempt):

3> [4383/4384] UE4Editor-UnrealEd.dll
3> Creating library C:\Unreal\Engine\Intermediate\Build\Win64\UE4Editor\Development\UnrealEd\UE4Editor-UnrealEd.suppressed.lib and object C:\Unreal\Engine\Intermediate\Build\Win64\UE4Editor\Development\UnrealEd\UE4Editor-UnrealEd.suppressed.exp
3>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.MakeFile.Targets(51,5): error MSB3073: The command "..\..\Build\BatchFiles\Rebuild.bat -Target="UE4Editor Win64 Development" -Target="ShaderCompileWorker Win64 Development -Quiet" -WaitMutex -FromMsBuild" exited with code -1.
3>Done building project "UE4.vcxproj" -- FAILED.
========== Rebuild All: 2 succeeded, 1 failed, 0 skipped ==========

In step 3 of the Unreal Engine, there is a note that says: "Keep the Unreal Engine folder as close as C:\\ as you can because if the path exceeds a certain length then Setup.bat will return errors in step 3." My path is C:\Unreal so I don't think that is the issue. 

When I try to run Engine\Binaries\Win64\UE4Editor.exe as a test in step 4 it comes up with the error shown in the attachment. 
<img width="303" alt="error" src="https://user-images.githubusercontent.com/106710525/172257792-88945f23-9798-4c63-9295-6fa4d4647fc2.PNG">

Looking for advice on what the issue is and how to troubleshoot. Thanks

**Comment by MoolixLin** [comment]
The main problem is the folder path length, try to shorten your path like this "E:/Project/UnrealEngine----->E:/UnrealEngine"

**Comment by JohnnyZZhao** [comment]
Hi I am building the UE4Editor on Windows by Visual Studio 2022, the build fails with the same error. Severity
`The command "..\..\Build\BatchFiles\Build.bat -Target="UE4Editor Win64 Development" -Target="ShaderCompileWorker Win64 Development -Quiet" -WaitMutex -FromMsBuild" exited with code 6.	UE4	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.MakeFile.Targets`	44	
Not sure if it's caused by the folder path, the path is C:\Project\Engine\Binaries
